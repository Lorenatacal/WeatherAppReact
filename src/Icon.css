.Icon-logo {
    animation: App-logo-spin infinite 30s linear;
    height: 90px;
  }