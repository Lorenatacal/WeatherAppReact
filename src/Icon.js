import React, { Component } from 'react';
import './Icon.css';

class Icon extends React.Component {
    constructor (props) {
        super(props)
        this.state = {
            rainIcon: "http://icons.iconarchive.com/icons/large-icons/large-weather/256/rain-icon.png",
            snowIcon: "http://www.free-icons-download.net/images/blue-snow-icon-38706.png",
            sunIcon: "http://www.pngmart.com/files/3/Sun-PNG-Photos.png",
            windIcon: "http://moziru.com/images/wind-clipart-weather-symbol-11.png",
            cloudIcon: "https://vignette.wikia.nocookie.net/farmville/images/5/57/Puffy_Cloud-icon.png/revision/latest?cb=20121010151231",
            fogIcon: "http://icongal.com/gallery/download/43079/256/png",
            mixtIcon: "https://images.onlinelabels.com/images/clip-art/Anonymous/Anonymous_simple_weather_symbols_13.png",
        };
    }
    render() {
        if(this.props.description.includes("rain")) {
            return (
                <div>
                    <img src={this.state.rainIcon} className="Icon-logo"/>
                </div>
            )
        } else if(this.props.description.includes("snow")) {
            return (
                <div>
                    <img src={this.state.snowIcon} className="Icon-logo"/>
                </div>
            )
        } else if(this.props.description.includes("sun")) {
            return (
                <div>
                    <img src={this.state.sunIcon} className="Icon-logo"/>
                </div>
            )
        } else if(this.props.description.includes("wind")) {
            return (
                <div>
                    <img src={this.state.windIcon} className="Icon-logo"/>
                </div>
            )
        } else if(this.props.description.includes("clouds")) {
            return (
                <div>
                    <img src={this.state.cloudIcon} className="Icon-logo"/>
                </div>
            )
        } else if(this.props.description.includes("fog")) {
            return (
                <div>
                    <img src={this.state.fogIcon} className="Icon-logo"/>
                </div>
            )
        } else {
            return (
                <div>
                    <img src={this.state.mixtIcon} className="Icon-logo"/>
                </div>
            )
        }
    }
}

export default Icon;