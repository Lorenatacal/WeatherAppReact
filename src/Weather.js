import React, { Component } from 'react';
import axios from 'axios';
import kelvinToCelsius from 'kelvin-to-celsius';
import kelvinToFahrenheit from 'kelvin-to-fahrenheit';
import styled from 'styled-components';
import Icon from './Icon';

const WeatherContainer = styled.div`
    border-radius: 20px;
    padding: 0.25em;
    margin: 2em;
    margin-left: 25px;
    margin-right: 25px;
    border: 1px solid lightgrey;
    align: center;
    text-align: center;
    font-family: arial;
`;

const StyledButton = styled.button`
    border-radius: 5px;
    padding: 0.25em;
    margin: 1em;
    border: 0.5px, #007bff;
    background: #007bff;
    color: white;
    text-align: center;
    font-size: 1.1rem;
    font-family: 'Merriweather', serif;
`

const StyledLocation = styled.h2`
    font-family: 'Source Serif Pro', serif;
`

const StyledText = styled.p`
    font-family: 'Merriweather', serif;
    font-size:  1.1rem;
`


class Weather extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            response: false,
            name: '',
            country: '',
            description: '',
            clouds: 0,
            icon: '',
            tempCelsiuse: 0,
            tempCelsiuseI: 0,
            tempFahrenheit: 0,
            minTempCelsiuse: 0,
            minTempFahrenheit: 0,
            maxTempCelsiuse: 0,
            maxTempFahrenheit: 0,
            wind: 0,
            humidity:0,
            showCelsiuse: true,
        };
    }

    handleClickFahrenheit = () => {
        this.setState({
            showCelsiuse: false,
        });
    }

    handleClickCelsiuse = () => {
        this.setState({
            showCelsiuse: true,
        });
    }

    render() {
        if (this.props.coords && !this.state.response){
            axios.get('https://api.openweathermap.org/data/2.5/weather?', {
                params: {
                    lat: this.props.coords.latitude,
                    lon: this.props.coords.longitude,
                    appid: process.env.REACT_APP_WEATHER_API_KEY,
                }
            })
            .then((response) => {
                this.setState({
                    response: true,
                    name: response.data.name,
                    country: response.data.sys.country,
                    description: response.data.weather[0].description,
                    tempCelsiuse: kelvinToCelsius(response.data.main.temp),
                    tempFahrenheit: kelvinToFahrenheit(response.data.main.temp),
                    minTempCelsiuse: kelvinToCelsius(response.data.main.temp_min),
                    minTempFahrenheit: kelvinToFahrenheit(response.data.main.temp_min),
                    maxTempCelsiuse: kelvinToCelsius(response.data.main.temp_max),
                    maxTempFahrenheit: kelvinToFahrenheit(response.data.main.temp_max),
                    clouds: response.data.clouds.all,
                    wind: response.data.wind.speed,
                    humidity: response.data.main.humidity,
                })
            })
        }

                return(
                    <WeatherContainer>
                        <StyledLocation> Today in: {this.state.name}, {this.state.country}</StyledLocation>
                        <Icon description={this.state.description}/>
                        <StyledText> We have {this.state.description}, with {this.state.clouds}% clouds</StyledText>
                        {
                            this.state.showCelsiuse ?  
                                <StyledText>The temperature is: {this.state.tempCelsiuse}°C, with minimum of {this.state.minTempCelsiuse}°C and a maximum of {this.state.maxTempCelsiuse}°C</StyledText> : 
                                <StyledText> The temperature is: {this.state.tempFahrenheit}°F, with minimum of {this.state.minTempFahrenheit}°F and a maximum of {this.state.maxTempFahrenheit}°F</StyledText>
                        }
                        <StyledText> The wind is blowing with: {this.state.wind}m/s and we have a humidity of: {this.state.humidity}%</StyledText>
                        <StyledButton onClick={this.handleClickFahrenheit}>Change °F</StyledButton>
                        <StyledButton onClick={this.handleClickCelsiuse}>Change °C</StyledButton>
                    </WeatherContainer>
                )
    }
}


export default Weather;

