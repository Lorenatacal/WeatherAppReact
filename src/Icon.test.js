import React, { Component } from 'react';
import axios from 'axios';
import Icon from './Icon';
import Weather from './Weather';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import MockAdapter from 'axios-mock-adapter';

Enzyme.configure({ adapter: new Adapter() });

test("Icon should show render rainIcon when the description includes the word rain", () => {
    // create description mocked property
    const description = "rain";
    // create a wrapper around the Icon componenent
    const wrapper = Enzyme.shallow(<Icon description={description} />);
    // take a shanpshot and compare result
    expect(toJson(wrapper)).toMatchSnapshot();
});

test("Icon should render snowIcon when the description includes the word snow)", () => {
    const description = "snow";
    const wrapper = Enzyme.shallow(<Icon description={description} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test("Icon should render sunIcon when the description includes the word sun", () => {
    const description="sun";
    const wrapper = Enzyme.shallow(<Icon description={description} />);
    expect(toJson(wrapper)).toMatchSnapshot();
})