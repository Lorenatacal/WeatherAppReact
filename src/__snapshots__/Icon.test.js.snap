// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render cloudIcon when the description includes the word cloud 1`] = `
<div>
  <img
    className="Icon-logo"
    src="https://vignette.wikia.nocookie.net/farmville/images/5/57/Puffy_Cloud-icon.png/revision/latest?cb=20121010151231"
  />
</div>
`;

exports[`Icon should render snowIcon when the description includes the word snow) 1`] = `
<div>
  <img
    className="Icon-logo"
    src="http://www.free-icons-download.net/images/blue-snow-icon-38706.png"
  />
</div>
`;

exports[`Icon should render sunIcon when the description includes the word sun 1`] = `
<div>
  <img
    className="Icon-logo"
    src="http://www.pngmart.com/files/3/Sun-PNG-Photos.png"
  />
</div>
`;

exports[`Icon should render windIcon when the description includes the word wind 1`] = `
<div>
  <img
    className="Icon-logo"
    src="http://moziru.com/images/wind-clipart-weather-symbol-11.png"
  />
</div>
`;

exports[`Icon should show render rainIcon when the description includes the word rain 1`] = `
<div>
  <img
    className="Icon-logo"
    src="http://icons.iconarchive.com/icons/large-icons/large-weather/256/rain-icon.png"
  />
</div>
`;
